aequilibrae/paths/path_file_saving.pyx

```diff
@@ -0,0 +1,74 @@
+# distutils: language = c++
+
+from libcpp.vector cimport vector
+from libcpp.string cimport string
+
+import pyarrow as pa
+cimport pyarrow as pa
+
+# need to decide or make optional which format we want
+import pyarrow.parquet as pq
+import pyarrow.feather as feather
+
+import numpy as np
+cimport numpy as np
+np.import_array()
+
+@cython.wraparound(False)
+@cython.embedsignature(True)
+@cython.boundscheck(False) # turn of bounds-checking for entire function
+cpdef void save_path_file(long origin_index,
+                          long num_links,
+                          long zones,
+                          long long [:] pred,
+                          long long [:] conn,
+                          string path_file,
+                          string index_file,
+                          bool write_feather):
+
+    cdef long long class_, node, predecessor, connector, ctr
+    cdef vector[long long] path_data
+    # could make this an ndarray and not do the conversion, we know the size of the index array is zones
+    cdef vector[long long] size_of_path_arrays
+    cdef long long* temp_data
+    cdef long long* temp_data_ind
+
+    cdef np.npy_intp dims[1]
+    cdef np.ndarray[np.longlong_t, ndim=1] numpy_array
+    cdef np.npy_intp dims_ind[1]
+    cdef np.ndarray[np.longlong_t, ndim=1] numpy_array_ind
+
+    for node in range(zones):
```

Comment:
This section would be inside a 

`with nogil:`

block

Reply:
yes, good catch